### PR TITLE
Ltac2: add primitives to build and compare Ltac2.Init.cast 

### LINKED ISSUE #11837: Ltac2 should expose the contents of the cast type
#### Description of the problem

I want to be able to build terms with, e.g., `vm_cast`, but `cast` is opaque:
https://github.com/coq/coq/blob/901cbfab468efa868e3838c2009ac09978ee661a/user-contrib/Ltac2/Init.v#L26

cc @ppedrot 

#### Coq Version

master

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix https://github.com/coq/coq/issues/11837

### CODE REVIEW

File: user-contrib/Ltac2/Constr.v [hunk 1]
```diff
@@ -27,13 +27,15 @@ Ltac2 Type case_invert := [
 | CaseInvert (constr array)
 ].
 
+Ltac2 Type cast_kind := [ VMCast | NativeCast | DefaultCast ].
```
[the remark this answers is not in the document]
Comment (SkySkimmer) [reply user-contrib/Ltac2/Constr.v:30]: This is part of the low level accessor `kind` already, no need to stack more layers IMO

File: doc/changelog/06-Ltac2-language/17468-tac2-cast.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- **Added:**
+  the cast kind type now exposes its constructors
```
Comment (jfehrle) [inline doc/changelog/06-Ltac2-language/17468-tac2-cast.rst:2]:
```suggestion
  The cast kind type now exposes its constructors
```

**Comment by SkySkimmer** [comment]
Last push was just the changelog change so https://gitlab.com/coq/coq/-/pipelines/827877810 is still a valid CI run

File: user-contrib/Ltac2/Constr.v (hunk 1 above)
[the remark this answers is not in the document]
Comment (SkySkimmer) [reply user-contrib/Ltac2/Constr.v]: changed to expose less as you insist

**Review by ppedrot — APPROVED** [review]

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**